build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -120,6 +121,33 @@ private void configureBwcProject(Project project, BwcVersions.UnreleasedVersionI
             buildBwcTaskProvider,
             "assemble"
         );
+
+        // only run this section for projects after the release of the stable API
+        if (bwcVersion.get().onOrAfter(Version.fromString("8.7.0"))) {
```

Comment:
nit: I would do a `return` from the method here in this case, so that the actual code below isn't nested so deep.